src/dao/admin.js

```diff
@@ -0,0 +1,294 @@
+const { Organization, OrgContent } = require('../models/orgModels');
+const { validate } = require('uuid');
+const { Sequelize } = require('sequelize');
+const { IdentityProvider } = require('../models/identityProvider');
+
+const createOrganization = async (orgData) => {
+    try {
+        const organization = await Organization.create({
+            ORG_NAME: orgData.orgName,
+            BUSINESS_OWNER: orgData.businessOwner,
+            BUSINESS_OWNER_CONTACT: orgData.businessOwnerContact,
+            BUSINESS_OWNER_EMAIL: orgData.businessOwnerEmail
+        });
+        return organization;
+    } catch (error) {
+        if (error instanceof Sequelize.UniqueConstraintError) {
+            throw error;
+        }
+        throw new Sequelize.DatabaseError(error);
+    }
+};
+
+const getOrganization = async (param) => {
+
+    const isUUID = validate(param);
+    const condition = isUUID ? { ORG_ID: param } : { ORG_NAME: param };
+
+    try {
+        const organization = await Organization.findOne({ where: condition });
+
+        if (!organization) {
+            throw new Sequelize.EmptyResultError('Organization not found');
+        }
+
+        return organization;
+    } catch (error) {
+        if (error instanceof Sequelize.EmptyResultError) {
+            throw error;
+        }
+        throw new Sequelize.DatabaseError(error);
+    }
+};
+
+const updateOrganization = async (orgData) => {
+    try {
+        const [updatedRowsCount, updatedOrg] = await Organization.update(
+            {
+                ORG_NAME: orgData.orgName,
+                BUSINESS_OWNER: orgData.businessOwner,
+                BUSINESS_OWNER_CONTACT: orgData.businessOwnerContact,
+                BUSINESS_OWNER_EMAIL: orgData.businessOwnerEmail
+            },
+            {
+                where: { ORG_ID: orgData.orgId },
+                returning: true
+            }
+        );
+
+        if (updatedRowsCount < 1) {
```

Comment:
This can be simplified as follows.

```js
if (updatedRowsCount < 1) {
    throw new Sequelize.EmptyResultError('Organization not found');
}
return [updatedRowsCount, updatedOrg];
```

Reply:
changed